packages/main/src/Calendar.ts

```diff
@@ -319,6 +332,38 @@ class Calendar extends CalendarPart {
 		this.secondaryCalendarType && this._setSecondaryCalendarTypeButtonText();
 	}
 
+	extractDateModes():string[] {
+		const format = (this.getFormat() as any);
+		if (!format) {
+			return ["day", "month", "year"];
+		}
+		const aPatternSymbolTypes = format.aFormatArray.map((oPatternSymbolSettings:{type:string}) => {
+			return oPatternSymbolSettings.type.toLowerCase();
+		});
+		const bDay = aPatternSymbolTypes.indexOf("day") >= 0;
```

Comment:
I know this was copied from OpenUI5, but still:
 - do not use hungarian notation
 - do not use `indexOf` but `includes`